com.unity.netcode.gameobjects/Tests/Runtime/Messaging/DisconnectReasonTests.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+using Unity.Netcode.TestHelpers.Runtime;
+
+namespace Unity.Netcode.RuntimeTests
+{
+    public class DisconnectReasonObject : NetworkBehaviour
+    {
+
+    }
+
+    public class DisconnectReasonTests : NetcodeIntegrationTest
```

Comment:
I don't see any tests for setting reason in connection approval.

Reply:
That is a good point and we should have some way for users to set that.
Just thinking about the context of this specific PR and it is a good question to mull-over:
"Is a client being "disconnected" if it hasn't really "connected" yet?"

Wondering if there needs to be a separate "connection refused/denied" PR?

Reply:
It IS already part of this PR though
https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/pull/2280/files#diff-d52d5b5b61ed193efdd8dc579c9919ac1493426df078c04d0aa0a4bd88cfaf96R456
My comment was there's no test for that new API

Reply:
My bad... the NetworkManagerHooks threw me off...
Yeah it is... so there should be a test for that too.  👍

Reply:
@SamuelBellomo that's a very valid point. Please see https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/pull/2280/commits/3030a05f0a5ff0adb0d0022b2942829b1ea04026 
Also consider approving :-)